bin/techsupport

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash -e
+
+ROOT=$(realpath $(dirname $0)/..)
+cd $ROOT
+
+TMPDIR=/tmp/techsupport
+mkdir -p $TMPDIR
+
+echo Collecting system information
+function save_to_file {
+  cmd=$1
+  outfile=${cmd// /_} # cleaner form of filename
+  outfile=${outfile//\-/}
+  $1 > $TMPDIR/$outfile.txt 2>/dev/null
+}
+save_to_file "sudo ovs-vsctl show"
+save_to_file "lsb_release -a"
+save_to_file "ip route"
+save_to_file "ip link show"
+save_to_file "arp -n"
+save_to_file "git describe --dirty"
+save_to_file "git status"
+save_to_file "git remote -v"
+save_to_file "git log -n5"
+
+if [ -d inst ]; then
+    echo Collecting runtime info from inst
+    cp -r inst $TMPDIR
+fi
+
+if [ -d local ]; then
+    echo Collecting config from local
+    cp -r local $TMPDIR
+    # clear out any private keys and key_ids (eg: in gcp_creds) if they exist
+    find $TMPDIR/local -name \*.json | xargs sed -i 's/\"private_key.*//' || true
+fi
+
+if [ -f local/system.conf ]; then
+    source local/system.conf
+fi
+
+# If an external switch was set up, try pinging it and saving that capture
+if [ -n "$ext_ctrl" ] && [ -n "$ext_addr" ]; then
+    echo Saving packet capture of ping to external switch
+    sudo timeout 4s tcpdump -i $ext_ctrl -w $TMPDIR/ping_switch.pcap &>/dev/null &
```

Comment:
this needs to be n+1 where n is the number of pings. Ping takes ~(n-1) time, but then you add one. So, right now you're still on the harry edge of the race condition.

Reply:
updated. 5 seconds of tcpdump, 1sec of sleep, 3 sec of pings, so we are at 1 second of wiggle room now